suite-native/discovery/src/discoverySelectors.ts

```diff
@@ -45,3 +71,84 @@ export const selectDiscoveryAccountsAnalytics = (
             };
         }),
     );
+
+export const selectNetworksWithUnfinishedDiscovery = (
+    state: DeviceRootState &
+        AccountsRootState &
+        FeatureFlagsRootState &
+        DiscoveryConfigSliceRootState,
+    areTestnetsEnabled: boolean,
+) => {
+    const enabledNetworkSymbols = selectEnabledDiscoveryNetworkSymbols(state);
+    const accounts = selectDeviceAccounts(state);
+    const supportedNetworks = selectDiscoverySupportedNetworks(state, areTestnetsEnabled);
+    const isCoinEnablingActive = selectIsFeatureFlagEnabled(
+        state,
+        FeatureFlag.IsCoinEnablingActive,
+    );
+
+    const enabledNetworks = supportedNetworks.filter(n =>
+        isCoinEnablingActive ? enabledNetworkSymbols.includes(n.symbol) : true,
+    );
+
+    return getNetworksWithUnfinishedDiscovery(enabledNetworks, accounts, ACCOUNTS_LIMIT);
+};
+
+//we should run discovery when there are network symbols with unfinished discovery
+export const selectShouldRunDiscoveryForDevice = (
+    state: DeviceRootState &
+        AccountsRootState &
+        FeatureFlagsRootState &
+        DiscoveryConfigSliceRootState,
+) => {
+    const isPortfolioTrackerDevice = selectIsPortfolioTrackerDevice(state);
+    if (isPortfolioTrackerDevice) {
+        return false;
+    }
+
+    const areTestnetsEnabled = selectAreTestnetsEnabled(state);
+    const networksWithUnfinishedDiscovery = selectNetworksWithUnfinishedDiscovery(
+        state,
+        areTestnetsEnabled,
+    );
+
+    return networksWithUnfinishedDiscovery.map(n => n.symbol).length > 0;
```

Comment:
The mapping seems redundant?
```suggestion
    return networksWithUnfinishedDiscovery.length > 0;
```

Reply:
b675fc3e18